### PR TITLE
`chore` Adding internal telemetry context

### PR DESCRIPTION
**Pull request by goergisn**
## Summary
- Adds internal `TelemetryContext` that specifies the version + platform

### CODE REVIEW

File: Adyen/Core/AdyenContext/AdyenContext.swift [hunk 1]
```diff
@@ -19,16 +19,20 @@ public final class AdyenContext: PaymentAware {
     public private(set) var payment: Payment?
 
     @_spi(AdyenInternal)
-    public let analyticsProvider: AnalyticsProviderProtocol
-
+    public var analyticsProvider: AnalyticsProviderProtocol
```
Comment (descorp) [inline Adyen/Core/AdyenContext/AdyenContext.swift:22]: Is this a good idea?
Maybe better to modify it via `AnalyticsConfiguration` ?

Comment (goergisn) [reply Adyen/Core/AdyenContext/AdyenContext.swift:22]: You mean moving the `TelemetryContext` to `AnalyticsConfiguration`? 
Could also be an option. No strong feelings on where it fits best 🤔 

What do you think @erenbesel

Comment (goergisn) [reply Adyen/Core/AdyenContext/AdyenContext.swift:22]: Please check https://github.com/Adyen/adyen-ios/pull/1473 for an alternative implementation

Comment (descorp) [reply Adyen/Core/AdyenContext/AdyenContext.swift:22]: I mean making `analyticsProvider` **var**

**Pull request closed without merging**